prisma/schema.prisma

```diff
@@ -0,0 +1,188 @@
+  passwordless_devices passwordless_devices @relation(fields: [device_id_hash], references: [device_id_hash], onDelete: Cascade, map: "passwordless_codes_ibfk_1")
+
+  @@index([device_id_hash], map: "device_id_hash")
+  @@index([created_at], map: "passwordless_codes_created_at_index")
+}
+
+model passwordless_devices {
+  device_id_hash     String               @id @db.Char(44)
+  email              String?              @db.VarChar(256)
+  phone_number       String?              @db.VarChar(256)
+  link_code_salt     String               @db.Char(44)
+  failed_attempts    Int                  @db.UnsignedInt
+  passwordless_codes passwordless_codes[]
+
+  @@index([email], map: "passwordless_devices_email_index")
+  @@index([phone_number], map: "passwordless_devices_phone_number_index")
+}
+
+model passwordless_users {
+  user_id      String  @id @db.Char(36)
+  email        String? @unique(map: "email") @db.VarChar(256)
+  phone_number String? @unique(map: "phone_number") @db.VarChar(256)
+  time_joined  BigInt  @db.UnsignedBigInt
+}
+
+model role_permissions {
+  role       String @db.VarChar(255)
+  permission String @db.VarChar(255)
+  roles      roles  @relation(fields: [role], references: [role], onDelete: Cascade, onUpdate: Restrict, map: "role_permissions_ibfk_1")
+
+  @@id([role, permission])
+  @@index([permission], map: "role_permissions_permission_index")
+}
+
+model roles {
+  role             String             @id @db.VarChar(255)
+  role_permissions role_permissions[]
+  user_roles       user_roles[]
+}
+
+model session_access_token_signing_keys {
+  created_at_time BigInt  @id @db.UnsignedBigInt
+  value           String? @db.Text
+}
+
+model session_info {
+  session_handle       String  @id @db.VarChar(255)
+  user_id              String  @db.VarChar(128)
+  refresh_token_hash_2 String  @db.VarChar(128)
+  session_data         String? @db.Text
+  expires_at           BigInt  @db.UnsignedBigInt
+  created_at_time      BigInt  @db.UnsignedBigInt
+  jwt_user_payload     String? @db.Text
+}
+
+model thirdparty_users {
+  third_party_id      String @db.VarChar(28)
+  third_party_user_id String @db.VarChar(128)
+  user_id             String @unique(map: "user_id") @db.Char(36)
+  email               String @db.VarChar(256)
+  time_joined         BigInt @db.UnsignedBigInt
+
+  @@id([third_party_id, third_party_user_id])
+}
+
+model user_metadata {
+  user_id       String @id @db.VarChar(128)
+  user_metadata String @db.Text
+}
+
+model user_roles {
+  user_id String @db.VarChar(128)
+  role    String @db.VarChar(255)
+  roles   roles  @relation(fields: [role], references: [role], onDelete: Cascade, onUpdate: Restrict, map: "user_roles_ibfk_1")
+
+  @@id([user_id, role])
+  @@index([role], map: "user_roles_role_index")
+}
+
+// End region: supertokens models
+
+// To be removed with sequelize
+model SequelizeMeta {
+  name String @id @unique(map: "name") @db.VarChar(255)
+}
+
```

Comment:
Do we really need all these models for supertokens? We are currently pointing a local instance to our database, but that's not the most common way of doing it. 

Some people might run it in separate databases (even our production stack), and most people would just use the free tier from https://supertokens.com/auth. 

We might be coupling it to a third-party service without benefits.

Reply:
Indeed, there is no need for modeling all of these super token tables.

I think that in the future we may want to access users' models on the backend, for example, if we decide to have something like granulate permissions for the server users. Since this is not currently on scope, I just removed all of it, together with the `SequelizeMeta` model.